packages/brick_offline_first_with_rest/lib/src/offline_queue/rest_offline_queue_client.dart

```diff
@@ -89,17 +99,26 @@ class RestOfflineQueueClient extends http.BaseClient {
       // Attempt to make HTTP Request
       final resp = await _inner.send(request);
 
-      if (cacheItem.requestIsPush && !reattemptForStatusCodes.contains(resp.statusCode)) {
-        final db = await requestManager.getDb();
-        // request was successfully sent and can be removed
-        _logger.finest('removing from queue: ${cacheItem.toSqlite()}');
-        await cacheItem.delete(db);
+      if (cacheItem.requestIsPush) {
+        if (!reattemptForStatusCodes.contains(resp.statusCode)) {
+          final db = await requestManager.getDb();
+          // request was successfully sent and can be removed
+          _logger.finest('removing from queue: ${cacheItem.toSqlite()}');
+          await cacheItem.delete(db);
+        } else {
+          _logger.finest(
+              'request failed, will be reattempted: ${cacheItem.toSqlite()}');
+          onReattemptableResponse?.call(request, resp);
+        }
```

Comment:
This will create a lot of logs for scenarios that want to disregard the response; it should be guarded
```suggestion
        } else if (onReattemptableResponse != null) {
          _logger.finer(
              'request failed, will be reattempted: ${cacheItem.toSqlite()}');
          onReattemptableResponse?.call(request, resp);
        }
```